src/main/java/org/commcare/cases/model/StorageIndexedTreeElementModel.java

```diff
@@ -0,0 +1,151 @@
+package org.commcare.cases.model;
+
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.core.model.instance.TreeElement;
+import org.javarosa.core.services.storage.IMetaData;
+import org.javarosa.core.services.storage.Persistable;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.javarosa.core.util.externalizable.ExtUtil;
+import org.javarosa.core.util.externalizable.ExtWrapList;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.Vector;
+
+/**
+ * A DB model for storing TreeElements such that particular attributes and
+ * elements are indexed and queryable using the DB.
+ *
+ * Indexed attributes/elements get their own table columns, and the rest of
```

Comment:
Woah. I legitimately didn't realize this would literally fit into the existing model set. That's actually pretty cool